include/fsm/fill_spill_merge.hpp

```diff
@@ -182,8 +182,8 @@ void FillSpillMerge(
   for(int d=1;d<(int)deps.size();d++){
     const auto &dep = deps.at(d);
     assert(dep.water_vol==0 || dep.water_vol<=dep.dep_vol);
-    assert(dep.water_vol==0 || (dep.lchild==NO_VALUE && dep.rchild==NO_VALUE) || (dep.lchild!=NO_VALUE && deps.at(dep.lchild).water_vol<dep.water_vol));
-    assert(dep.water_vol==0 || (dep.lchild==NO_VALUE && dep.rchild==NO_VALUE) || (dep.rchild!=NO_VALUE && deps.at(dep.rchild).water_vol<dep.water_vol));
+    assert(dep.water_vol==0 || (dep.lchild==NO_VALUE && dep.rchild==NO_VALUE) || (dep.lchild!=NO_VALUE && fp_le(deps.at(dep.lchild).water_vol,dep.water_vol)));
+    assert(dep.water_vol==0 || (dep.lchild==NO_VALUE && dep.rchild==NO_VALUE) || (dep.rchild!=NO_VALUE && fp_le(deps.at(dep.rchild).water_vol,dep.water_vol)));
```

Comment:
These seem a little worrisome, as though somewhere an `std::min` should have set things to exactly-equal or < ?

Reply:
It looks like we have this same assertion at the end of MoveWaterinDephier. Perhaps I should set it to the std::min there and then not change the assertion here?

Reply:
This approach is nice because it's a safe-guard against a too-big amount of water being added, whereas a pure `std::min()` could hide a bug. But `fp_le()` does allow watervol to exceed depvol. I could accept this, but at some point it might be worth going through and adding a function with contents like: `if(dep_vol<water_vol && water_vol<dep_vol+1e-6) water_vol=dep_vol;`